### PR TITLE
#1528 Portfolio overhaul and fixes

### PR DESCRIPTION
**Pull request by drakkein**

### CODE REVIEW

**Comment by drakkein** [comment]
@rudolfix @desfero I see that row is disappearing only when there is an error during wallet request. Can we have short discussion on Monday how we handle this? I guess we should cache wallet data to have access to last correct state in case of error.

File: app/components/shared/NewTable.tsx [hunk 1]
```diff
@@ -20,8 +20,13 @@ interface IPlaceholderTableRow {
...
+  | React.ReactElement<INewTableRow>
+  | React.ReactElement<INewTableRow>[]
+  | boolean;
+
 interface INewTable {
-  children: React.ReactElement<INewTableRow> | React.ReactElement<INewTableRow>[];
+  children: INewTableChildren | INewTableChildren[];
   placeholder?: TTranslatedString;
   className?: string;
   keepRhythm?: boolean;
```
Comment (AlexeyTuboltsev) [inline app/components/shared/NewTable.tsx]: can you please change the uniqueId() on the line 45? (it's not your code but we can't leave it like this ))

File: app/components/shared/NewTable.tsx [hunk 2]
```diff
@@ -20,8 +20,13 @@ interface IPlaceholderTableRow {
   numberOfCells: number;
 }
 
+type INewTableChildren =
+  | React.ReactElement<INewTableRow>
+  | React.ReactElement<INewTableRow>[]
+  | boolean;
```
Comment (AlexeyTuboltsev) [inline app/components/shared/NewTable.tsx]: I suggest to change this to null. Element or null is idiomatic react way to deal with conditional rendering

File: app/components/portfolio/PortfolioLayout.tsx [hunk 3]
```diff
@@ -201,6 +186,34 @@ const PortfolioLayout: React.SFC<TPortfolioLayoutProps> = ({ myAssets, pendingAs
...
+              />
+              <>{"-"}</>
+              <Link
+                to={withParams(externalRoutes.commitmentStatus, { walletAddress })}
+                target={"_blank"}
+              >
+                <FormattedMessage id="portfolio.section.your-assets.tokenholder-agreement" />
+              </Link>
+            </NewTableRow>
+          )}
```
Comment (AlexeyTuboltsev) [inline app/components/portfolio/PortfolioLayout.tsx]: if you change the conditional to {condition ? <NewTableRow> : null} 
you won't set newTableRow type to boolean, which is a bit strange and counterintuitive

File: app/components/portfolio/PortfolioLayout.tsx [hunk 4]
```diff
@@ -39,72 +46,50 @@ const getNeuReward = (equityTokenInt: BigNumber, equivEurUlps: BigNumber): strin
...
+              size="large"
+              moneyValue="100000000"
+              moneyChange={-20}
+              tokenValue="1000000"
+              tokenChange={20}
+            />
+          </Col>
+        </Row>
+      </>
+    )}
```
Comment (AlexeyTuboltsev) [inline app/components/portfolio/PortfolioLayout.tsx]: please add some top margin to the next element, if assets are not visible.

**Comment by rudolfix** [comment]
> @rudolfix @desfero I see that row is disappearing only when there is an error during wallet request. Can we have short discussion on Monday how we handle this? I guess we should cache wallet data to have access to last correct state in case of error.

@drakkein sure, I had a busy day today but tomorrow let's squeeze some time. around 2pm. probably. thanks!

**Pull request merged**